core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -4226,7 +4226,7 @@ public void assertSameReader(Searcher left, Searcher right) {
         List<LeafReaderContext> rightLeaves = ElasticsearchDirectoryReader.unwrap(right.getDirectoryReader()).leaves();
         assertEquals(rightLeaves.size(), leftLeaves.size());
         for (int i = 0; i < leftLeaves.size(); i++) {
-            assertSame(leftLeaves.get(i).reader(), rightLeaves.get(0).reader());
+            assertSame(leftLeaves.get(i).reader(), rightLeaves.get(i).reader());
```

Comment:
sneaky :)

Reply:
yeah suddenly we have a test that triggered it 💃